### PR TITLE
docs(concepts): add self-service to features

### PR DESCRIPTION
**Pull request by mffap**
Explains the self-service capabilities in ZITADEL

This doc is not perfect by any means, but a starting point to answer the questions around self-service capabilities. In the same docs I wanted to introduce the special aspects of ZITADEL where we not only provide CIAM self-service but also B2B self-service capabilities.

### CODE REVIEW

**Review by hifabienne — DISMISSED** [review]

File: docs/docs/concepts/features/selfservice.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
+---
+title: Self-Service
+---
+
+ZITADEL allows users to perform certain tasks themselves.
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:5]:
```suggestion
ZITADEL allows users to perform many tasks themselves.
```

File: docs/docs/concepts/features/selfservice.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+For these tasks we either provide an user interface, or the tasks can be initiated or completed through our APIs.
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:6]:
```suggestion
For these tasks we either provide an user interface, or the tasks can be initiated or completed through ZITADEL's APIs.
```

File: docs/docs/concepts/features/selfservice.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+It is important to understand that, depending on your use case, there will exist different user-types that want to perform different actions:  
+
+- `User` are the end-users of your application. Like with any CIAM solution, users should be able to perform tasks like register/join, update their profile, manage authenticators etc.There are certain actions that can be executed pre-login, yet others require the user to have a valid session.
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:10]:
```suggestion
- `Users` are the end-users of your application. Like with any CIAM solution, users should be able to perform tasks like register/join, update their profile, manage authenticators etc. There are certain actions that can be executed pre-login, yet others require the user to have a valid session.
```

File: docs/docs/concepts/features/selfservice.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+- `Manager` are users with a [special manager role within ZITADEL](/docs/concepts/structure/managers) and can perform administrative actions such as system configuration or granting access rights to users.
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:11]:
```suggestion
- `Managers` are users with a [special manager role within ZITADEL](/docs/concepts/structure/managers) and can perform administrative actions such as system configuration or granting access rights to users.
```

File: docs/docs/concepts/features/selfservice.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+:::info
+ZTIADEL covers the typical "CIAM" self-service capabilities as well as delegated access management for multi-tenancy scenarios. Please refer to the section [Managers](#managers).
+:::
+
+## Registration
+
+:::info
+You can pre-select a given organization by passing the scope `urn:zitadel:iam:org:domain:primary:{domainname}` in the authorization request.
+This will force users to register only with the specified organization.
+Also the branding and login settings (e.g. Social Login Providers) are directly shown to the user.
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:24]:
```suggestion
Furthermore the branding and login settings (e.g. Social Login Providers) are directly shown to the user.
```

File: docs/docs/concepts/features/selfservice.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+:::
+
+### Local account
+
+Allows unauthenticated and users, who are not yet in the given organization, to create an account (register) themselves.
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:29]:
```suggestion
Allows anonymous users and authenticated users, who are not yet in the given organization, to create an account (register) themselves.
```

File: docs/docs/concepts/features/selfservice.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+
+- Mandatory profile fields
+- Set password
+- Accept terms of service and privacy policy
+- User receives an email with an one-time code
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:34]:
```suggestion
- User receives an email with an one-time code for verfication purpose
```

File: docs/docs/concepts/features/selfservice.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+- User has to enter the one-time code to finish registration
+- User can re-quest a new one-time code
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:36]:
```suggestion
- User can re-request a new one-time code
```

File: docs/docs/concepts/features/selfservice.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+The user is prompted on first login to setup MFA.
+This step is mandatory in case MFA is enforced in the login policy.
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:39]:
```suggestion
This step can be made mandatory if MFA is enforced in the login policy.
```

File: docs/docs/concepts/features/selfservice.md [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+
+### Existing Identity / SSO / Social Login
+
+Allows unauthenticated and users, who are not yet in the given organization, to register with an external identity provider.
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:43]:
```suggestion
anonymous users and authenticated users, who are not yet in the given organization, to register with an external identity provider.
```

File: docs/docs/concepts/features/selfservice.md [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+An external identity provider can be a Social Login Provider or a pre-configured identity provider.
+
+- Information from the external identity provider is used to pre-fill the profile information
+- User can update the profile information
+- An account is created within ZITADEL and linked with the external identity provider
+
+#### Account Linking
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:50]: I think we should mention autoregister / autolinking as well here

File: docs/docs/concepts/features/selfservice.md [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+
+[This guide](/docs/guides/integrate/login-users) explains in more detail the login-flows for different application types.
+Human users are redirected to ZITADEL's login page and complete sign-in with the interactive login flow.
+It is important to understand that ZITADEL provides a hosted login page and the device of the users opens this login page in a browser, even on Native/Mobile apps.
+
+#### MFA / 2FA
+
+Users are automatically prompted to provide a second factor, when
+
+- Instance or organization [login policy](/docs/concepts/structure/policies#login-policy)
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:75]:
```suggestion
- Instance or organization [login policy](/docs/concepts/structure/policies#login-policy) is set
```

File: docs/docs/concepts/features/selfservice.md [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+- Requested by the client
+- A multi-factor is setup for the user
+
+When a multi-factor is required, but not set-up, then the user is requested to set-up an additional factor.
+
+#### FIDO Passkeys
+
+Users can select a button to initiate passwordless login or use a fall-back method (ie. login with username/password), if available.
+
+The passwordless login flow follows the FIDO 2 / WebAuthN standard.
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:85]:
```suggestion
The passwordless login flow follows the FIDO2 / WebAuthN standard.
```

File: docs/docs/concepts/features/selfservice.md [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+Briefly explained the following happens:
+
+- User selects button
+- User's device will ask the user to provide a gesture (eg, FaceID, Windows Hello, Fingerprint, PIN)
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:89]:
```suggestion
- User's device will ask the user to provide a gesture (e.g., FaceID, Windows Hello, Fingerprint, PIN)
```

File: docs/docs/concepts/features/selfservice.md [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+- The user is being redirected to the application
+
+With the introduction of passkeys the gesture can be provided on ANY of the user's devices.
+This is not strictly the device where the login flow is being executed (eg, push notification on a mobile device).
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:93]:
```suggestion
This is not strictly the device where the login flow is being executed (e.g., on a mobile device).
```

File: docs/docs/concepts/features/selfservice.md [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+- after successful login the user will be redirected to the application
+
+### Machines
+
+Machine accounts can't use an interactive login but require other means of authentication, such as privately-signed JWT or personal access tokens.
+Read more about [Service Users](/docs/guides/integrate/serviceusers) and recommended [OpenID Connect Flows](/docs/guides/integrate/oauth-recommended-flows#different-client-profiles).
+
+### Other Clients
+
+We currently do not expose the Login APIs.
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:111]:
```suggestion
We currently do not expose the Login API.
```

File: docs/docs/concepts/features/selfservice.md [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+Unauthenticated users can request a password reset after providing the loginname during the login flow.
+
+- User selects reset password
+- An email will be sent to the verified email address
+- User opens a link and has to provide a new password
+
+## Logout
+
+Users can terminate the session for all their users (logout).
+A client will implement this by calling the [specific endpoint](/docs/apis/openidoauth/endpoints#end_session_endpoint).
```
Comment (fforootd) [inline docs/docs/concepts/features/selfservice.md:136]:
```suggestion
A client can also implement this, by calling the [specific endpoint](/docs/apis/openidoauth/endpoints#end_session_endpoint).
```

**Review by fforootd — CHANGES_REQUESTED** [review]

File: docs/docs/concepts/features/selfservice.md [hunk 18]
```diff
@@ -0,0 +1,214 @@
...
+anonymous users and authenticated users, who are not yet in the given organization, to register with an external identity provider.
+An external identity provider can be a Social Login Provider or a pre-configured identity provider.
+
+- Information from the external identity provider is used to pre-fill the profile information
+- User can update the profile information
+- An account is created within ZITADEL and linked with the external identity provider
+
+#### Account Linking
+
+When you login with an external identity provider, and the user does not exist in ZITADEL, then the user is presented with two options:
```
Comment (mffap) [inline docs/docs/concepts/features/selfservice.md:52]:
```suggestion
When you login with an external identity provider, and the user does not exist in ZITADEL, then an autoregister flow is triggered. The user is presented with two options: 
```

File: docs/docs/concepts/features/selfservice.md [hunk 19]
```diff
@@ -0,0 +1,214 @@
...
+- Information from the external identity provider is used to pre-fill the profile information
+- User can update the profile information
+- An account is created within ZITADEL and linked with the external identity provider
+
+#### Account Linking
+
+When you login with an external identity provider, and the user does not exist in ZITADEL, then the user is presented with two options: 
+
+- Create a new account: A new account will be created as stated above
+- Link an existing account: The user is prompted to login with an existing [local account](#local-account). If successful, the existing identity from the external identity provider will be linked with the local account. A user can now login with either the local account or any of the linked external accounts.
```
Comment (mffap) [inline docs/docs/concepts/features/selfservice.md:55]:
```suggestion
- Autolinking: The user is prompted to login with an existing [local account](#local-account). If successful, the existing identity from the external identity provider will be linked with the local account. A user can now login with either the local account or any of the linked external accounts.
```

File: docs/docs/concepts/features/selfservice.md (hunk 11 above)
Comment (mffap) [reply docs/docs/concepts/features/selfservice.md:50]: Ok. Done.

**Review by fforootd — APPROVED** [review]

**Pull request merged**